java/com/fortech/internship/dao/CityDAOImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.fortech.internship.dao;
+
+import java.util.ArrayList;
+
+import java.util.List;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.fortech.internship.dto.CityDTO;
+import com.fortech.internship.model.Address;
+import com.fortech.internship.model.City;
+import com.fortech.internship.model.User;
+
+@Component
+@Repository
+@Transactional(propagation = Propagation.REQUIRED)
+public class CityDAOImpl implements CityDAO {
+
+	@Autowired
+	private SessionFactory sessionFactory;
+
+	@Override
+	public void createCity(CityDTO cityDTO) {
```

Comment:
you should pass City as a parameter, not CityDTO. The creation of the City object should be done in the Service Layer

Reply:
solved, please check